sdk/keyvault/Azure.Security.KeyVault.Secrets/tests/SecretsTests.cs

```diff
@@ -0,0 +1,259 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Azure.Security.KeyVault.Secrets;
+using NUnit.Framework;
+using Azure.Identity;
+
+namespace Azure.Security.KeyVault.Test
+{
+    public class SecretTests : KeyVaultTestBase
+    {
+        static void Main(string[] args)
+        {
+
+        }
+        public SecretTests()
+        {
+        }
+
+        [Test]
+        public async Task CredentialProvider()
+        {
+            var client = new Secrets.SecretClient(VaultUri, AzureCredential.Default);
+
+            Secret setResult = await client.SetAsync("CrudBasic", "CrudBasicValue1");
+
+            Secret getResult = await client.GetAsync("CrudBasic");
+
+            AssertSecretsEqual(setResult, getResult);
+
+            DeletedSecret deleteResult = await client.DeleteAsync("CrudBasic");
+
+            AssertSecretsEqual(setResult, deleteResult);
+        }
+
+        [Test]
+        public async Task CrudBasic()
+        {
+            var client = new SecretClient(VaultUri, AzureCredential.Default);
+
+            Secret setResult = await client.SetAsync("CrudBasic", "CrudBasicValue1");
+
+            Secret getResult = await client.GetAsync("CrudBasic");
+
+            Assert.AreEqual("CrudBasic", setResult.Name);
+            Assert.AreEqual(VaultUri, setResult.Vault);
+
+            AssertSecretsEqual(setResult, getResult);
+
+            getResult.Enabled = false;
+            SecretBase updateResult = await client.UpdateAsync(getResult);
+
+            AssertSecretsEqual(getResult, updateResult);
+
+            DeletedSecret deleteResult = await client.DeleteAsync("CrudBasic");
+
+            AssertSecretsEqual(updateResult, deleteResult);
+        }
+
+        [Test]
+        public async Task CrudWithExtendedProps()
+        {
+            var client = new SecretClient(VaultUri, AzureCredential.Default);
+
+            var secret = new Secret("CrudWithExtendedProps", "CrudWithExtendedPropsValue1")
+            {
+                ContentType = "password",
+                NotBefore = UtcNowMs() + TimeSpan.FromDays(1),
+                Expires = UtcNowMs() + TimeSpan.FromDays(90)
+            };
+
+            Secret setResult = await client.SetAsync(secret);
+
+            Assert.AreEqual("password", setResult.ContentType);
+
+            Secret getResult = await client.GetAsync("CrudWithExtendedProps");
+
+            AssertSecretsEqual(setResult, getResult);
+
+            DeletedSecret deleteResult = await client.DeleteAsync("CrudWithExtendedProps");
+
+            // remove the value which is not set on the deleted response
+            typeof(Secret).GetProperty(nameof(setResult.Value)).SetValue(setResult, null);
```

Comment:
Remove all usages of this reflection hack. If you need to avoid checking the value add a parameter to `AssertSecretsEqual`